tests/integration/features/bootstrap/FeatureContext.php

```diff
@@ -1103,6 +1103,36 @@ public function userSendsMessageToRoom($user, $message, $identifier, $statusCode
 		}
 	}
 
+	/**
+	 * @Then /^user "([^"]*)" shares rich-object "([^"]*)" "([^"]*)" '([^']*)' to room "([^"]*)" with (\d+)(?: \((v(1|2|3))\))?$/
+	 *
+	 * @param string $user
+	 * @param string $type
+	 * @param string $id
+	 * @param string $metaData
+	 * @param string $identifier
+	 * @param string $statusCode
+	 * @param string $apiVersion
+	 */
+	public function userSharesRichObjectToRoom($user, $type, $id, $metaData, $identifier, $statusCode, $apiVersion = 'v1') {
+		$this->setCurrentUser($user);
+		$this->sendRequest(
+			'POST', '/apps/spreed/api/' . $apiVersion . '/chat/' . self::$identifierToToken[$identifier] . '/share',
+			new TableNode([
+				['objectType', $type],
+				['objectId', $id],
+				['metaData', $metaData],
+			])
+		);
+		$this->assertStatusCode($this->response, $statusCode);
+		sleep(1); // make sure Postgres manages the order of the messages
```

Comment:
postgres only sorts results while sleeping ? :open_mouth:

Reply:
Don't like that very much either

Reply:
No, postgres uses journaling and without a sleep 2 posts from the same second might be sorted wrongly when you dont explicitly sort by id, it's just copied from the normal message posting

Reply:
https://github.com/nextcloud/spreed/pull/648

Reply:
This is in integration tests, there is no way around it apart from making tests unreliable, ignore and carry on please

Reply:
Okay, now I understand the comment wording better :+1: